cmd/cdi-cloner/header/sizeheader.go

```diff
@@ -0,0 +1,61 @@
+package main
+
+import (
+	"encoding/binary"
+	"flag"
+	"os"
+	"strconv"
+
+	"k8s.io/klog"
+)
+
+var (
+	namedPipe  *string
+	sizeString *string
+)
+
+func init() {
+	sizeString = flag.String("size", "0", "The size to write to the named pipe")
+	namedPipe = flag.String("pipedir", "nopipedir", "The name and directory of the named pipe to write to")
+	flag.Parse()
+	klogFlags := flag.NewFlagSet("klog", flag.ExitOnError)
+	klog.InitFlags(klogFlags)
+	flag.CommandLine.VisitAll(func(f1 *flag.Flag) {
+		f2 := klogFlags.Lookup(f1.Name)
+		if f2 != nil {
+			value := f1.Value.String()
+			f2.Value.Set(value)
+		}
+	})
+}
+
+func main() {
+	defer klog.Flush()
+	klog.V(1).Infoln("Starting cloner target")
+
+	if *namedPipe == "nopipedir" {
+		klog.Errorln("Missing named pipe flag")
+		os.Exit(1)
+	}
+	if *sizeString == "0" {
+		klog.Errorln("Missing size flag")
+		os.Exit(1)
+	}
+
+	size, err := strconv.ParseUint(*sizeString, 10, 64)
+	if err != nil {
+		klog.Errorf("Unable to parse size string: %+v", err)
+		os.Exit(1)
+	}
+
+	out, err := os.OpenFile(*namedPipe, os.O_WRONLY, os.ModeNamedPipe)
+	if err != nil {
+		klog.Errorf("Unable to open named pipe: %+v", err)
+		os.Exit(1)
+	}
+	defer out.Close()
+
+	b := make([]byte, 8)
+	binary.LittleEndian.PutUint64(b, size)
+	out.Write(b)
```

Comment:
I think it would be wise to check the values returned by this function

Reply:
Switched to writing with print, so this whole file goes away.